sklearnex/preview/ensemble/_forest.py

```diff
@@ -415,6 +575,17 @@ def _onedal_fit_ready(self, patching_status, X, y, sample_weight):
                     "`max_sample=None`."
                 )
 
+            if (
```

Comment:
This check was written into all patching checks for fit, this can be moved into a more central location to minimize duplication.